src/mame/sega/sega_beena.cpp

```diff
@@ -71,131 +102,2074 @@
+		}
+		// Afterwards apply `load16_word_swap`, regardless of host's endianness,
+		// since this reflects how ROM data was stored, not the CPU's architecture.
+		uint16_t *rom16 = reinterpret_cast<uint16_t *>(m_cart->get_rom_base());
+		for (size_t i = 0; i < size / 2; i++) {
+			rom16[i] = swapendian_int16(rom16[i]);
+		}
+	}
+
 	return std::make_pair(std::error_condition(), std::string());
 }
 
-static INPUT_PORTS_START( sega_beena )
-INPUT_PORTS_END
+CROSSHAIR_MAPPER_MEMBER(sega_beena_state::pen_y_mapper)
+{
+	// TODO: Either remove or adapt for Storyware layout
+	return linear_value;
+}
 
 
-uint32_t sega_beena_state::screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
+class tvochken_state : public sega_9h0_0008_state
 {
-	return 0;
+public:
+	tvochken_state(const machine_config &mconfig, device_type type, const char *tag)
+		: sega_9h0_0008_state(mconfig, type, tag)
+		, m_io_buttons(*this, "BUTTONS")
+		, m_io_cards(*this, "CARDS")
+		{ }
+
+	void tvochken(machine_config &config);
+
+	virtual uint32_t io_expansion_r() override;
+
+private:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+	virtual void install_game_rom() override;
+
+	required_ioport m_io_buttons;
+	required_ioport m_io_cards;
+	uint8_t m_card_previous_input;
+
+	enum card_state : uint8_t
+	{
+		IDLE = 0,
+		START_WRITE_DATA,
+		WRITE_DATA,
+	};
+	uint8_t m_card_i;
+	uint16_t m_card_data;
+	uint8_t m_card_data_i;
+	uint8_t m_card_state;
+	uint8_t m_card_hold_i;
+	uint8_t m_card_status;
+};
+
+void tvochken_state::machine_start()
+{
+	sega_9h0_0008_state::machine_start();
+
+	save_item(NAME(m_card_i));
+	save_item(NAME(m_card_data));
+	save_item(NAME(m_card_data_i));
+	save_item(NAME(m_card_hold_i));
+	save_item(NAME(m_card_state));
+	save_item(NAME(m_card_status));
 }
 
-void sega_beena_state::sega_beena(machine_config &config)
+void tvochken_state::machine_reset()
 {
-	ARM7_BE(config, m_maincpu, 81'000'000);
-	m_maincpu->set_addrmap(AS_PROGRAM, &sega_beena_state::beena_arm7_map);
+	sega_9h0_0008_state::machine_reset();
 
-	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
-	screen.set_refresh_hz(60);
-	screen.set_vblank_time(ATTOSECONDS_IN_USEC(0));
-	screen.set_size(640, 480);
-	screen.set_visarea(0, 640-1, 0, 480-1);
-	screen.set_screen_update(FUNC(sega_beena_state::screen_update));
+	m_card_previous_input = 0;
 
-	SPEAKER(config, "lspeaker").front_left();
-	SPEAKER(config, "rspeaker").front_right();
+	m_card_i = 0;
+	m_card_data = 0;
+	m_card_data_i = 0;
+	m_card_hold_i = 0;
+	m_card_state = IDLE;
+	m_card_status = 0;
+}
 
-	GENERIC_CARTSLOT(config, m_cart, generic_plain_slot, "sega_beena_cart");
-	m_cart->set_width(GENERIC_ROM16_WIDTH);
-	m_cart->set_device_load(FUNC(sega_beena_state::cart_load));
-	m_cart->set_must_be_loaded(true);
+void tvochken_state::tvochken(machine_config &config)
+{
+	sega_9h0_0008_state::sega_9h0_0008(config);
+}
```

Comment:
Do you really need this function?  You can use the base one directly if it isn’t different.  At any rate, you don’t need to qualify the base member call with the class name here since it isn’t obscured by an identically named member in the derived class.

Reply:
Not right now, but we can reintroduce it later on if needed.